master/flock_server/static/flock-mpi.js

```diff
@@ -508,3 +510,50 @@ mpi.ireduce = async function (sendArr, op, comm) {
         return local;
     }
 }
+
+
+/**
+ * Each process sends the contents of its sendbuffer to the root process. The root process
+ * receives the messages and stores them in rank order.
+ * 
+ * @async
+ * @function mpi.igather
+ * 
+ * @param {array} sendArr - Array to 
+ * @param {function} root - The rank of the root node for the gather operation
+ * @param {string} comm - Name of the communication group to operate under
+ * 
+ * @returns {serializable} - Result of the gather is given to the node with rank === 0, other nodes will receive error codes
```

Comment:
I would rather say 'status codes' than 'error codes' since we will be returning 200s as well